modules/product/src/main/java/com/opengamma/strata/product/SecurityProduct.java

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright (C) 2016 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.product;
+
+import com.opengamma.strata.basics.currency.Currency;
+import com.opengamma.strata.basics.market.ReferenceData;
+
+/**
+ * The product details of a security.
+ * <p>
+ * A {@code SecurityProduct} includes all the financial details of a security.
+ * It will include any underlying securities.
+ * For example, a bond future option would include the underlying bond future
+ * and the basket of bonds.
+ * <p>
+ * The product of a security is distinct from the {@linkplain ReferenceSecurity security}.
+ * The security, referred to using a {@link SecurityId}, does not include underlying securities.
+ * <p>
+ * Implementations of this interface must be immutable beans.
+ */
+public interface SecurityProduct
```

Comment:
My discussions indicated that it was important to retain the direct link in the trades to the product. It also became apparent that only one trade class covers all securities, thus it also doesn't create too much code. (Unlike OTCs, there is a choice in representing security-based trades, and the design reflects that)